pkg/util/metrics_helper.go

```diff
@@ -150,6 +150,21 @@ func (d MetricFamiliesPerUser) SendSumOfCountersPerUser(out chan<- prometheus.Me
 	}
 }
 
+// SendSumOfCountersPerUserWithLabels provides metrics with the provided label names on a per-user basis. This function assumes that `user` is the
+// first label on the provided metric Desc
+func (d MetricFamiliesPerUser) SendSumOfCountersPerUserWithLabels(out chan<- prometheus.Metric, desc *prometheus.Desc, metric string, labelNames ...string) {
+	for user, userMetrics := range d {
+		metricsPerLabelValue := getMetricsWithLabelNames(userMetrics[metric], labelNames)
+		for _, mlv := range metricsPerLabelValue {
+			for _, m := range mlv.metrics {
```

Comment:
The function name is `SendSumOfCountersPerUserWithLabels()` so if you have more than one `mlv.metrics` I would expect it sum them and the writes to `out` only 1 metric. On the contrary, I think with the current logic we will have clashing series (same exact series reported multiple times) and should be fixed. This case should be covered with a unit test too.

_ SendSumOfGaugesPerUserWithLabels_

Reply:
👍